compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -1734,16 +1734,24 @@ object Objects:
       if cls.isAllOf(Flags.JavaInterface) then Bottom
       else evalType(tref.prefix, thisV, klass, elideObjectAccess = cls.isStatic)
 
+  val mutateErrorSet: mutable.Set[(ClassSymbol, ClassSymbol)] = mutable.Set.empty
```

Comment:
Here we have a global mutable object, which is a concern. We can put it in a _contextual object_, i.e. to pass the state as method pararmeters.

Reply:
I changed `objects` to be a class as discussed.